elastic/apm-server
# Fix agent metrics

## Motivation/summary

Add omitted agent metrics, and fix names of agent metrics.

Also, if an agent sends "type" or "unit" for internal metrics, strip those from the metrics to ensure "_metric_descriptions" isn't set in the metricset document. This is set in the application metrics data stream for dynamic mapping; it is not used in internal metrics.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Run APM Server
2. Run all opbeans applications, pointed at APM Server (e.g. using apm-integration-testing with `--all-opbeans`)
3. Wait at least 1 minute to ensure all agents have sent some metrics
4. Ensure there are no service-specific application metrics data streams (`metrics-apm.app.*-*`)

Note that opbeans-python is instrumented with Prometheus, so it will generate `metrics-apm.app.opbeans_python-*` by default. Prometheus metrics can be disabled with the env var `ELASTIC_APM_PROMETHEUS_METRICS=False`, which should lead to the service-specific data stream no longer being populated.

## Related issues

Closes https://github.com/elastic/apm-server/issues/8263
